Source/CentralManagerRestoredState.swift

```diff
@@ -60,7 +60,10 @@ public struct CentralManagerRestoredState: CentralManagerRestoredStateType {
         let cbServices = arrayOfAnyObjects.flatMap { $0 as? CBService }
         #endif
 
-        return cbServices.map { Service(peripheral: centralManager.retrievePeripheral(for: $0.peripheral),
-                                        service: $0) }
+        return cbServices.map {
```

Comment:
You could kill two birds with one stone here and use `.compactMap` on the entire sequence, rather than chaining to it at the end
```suggestion
        return cbServices.compactMap {
            guard let cbPeripheral = $0.peripheral else { return nil }
            let peripheral = centralManager.retrievePeripheral(for: cbPeripheral)
            return Service(peripheral: peripheral, service: $0)
        }
```

Reply:
I have made the change you can have a look again